gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -164,7 +164,11 @@ public void close() throws IOException {
         this.job.killJob();
       }
     } finally {
-      super.close();
+      try {
+        cleanUpWorkingDirectory();
```

Comment:
`runWorkUnits` may not get a chance to be called if the logic fails before it. `close` is guaranteed to be called.